archive_v1/requirements.txt

```diff
@@ -0,0 +1,31 @@
+wandb~=0.12.9
+
+# models
+transformers>=4.20.1,<4.21.0
+torch>=1.6.0,<1.13.0
+
+# data
+# datasets>=2.3.2,<2.4.0
+git+https://github.com/huggingface/datasets  # dev version needed for running the training script
+ndjson>=0.3.1,<0.4.0
+
+## data cleaning
+spacy>=3.1.4,<3.5.0
+datasketch>=1.5.6,<1.6.0
+google==3.0.0  # safesearch filtering
+more_itertools>=8.12.0,<8.14.0
+pandas>=1.0.0,<2.0.0
+emoji>=1.7.0,<1.8.0
+luga>=0.2.5,<0.3.0
+langdetect>=1.0.9,<1.1.0
+
+# utility
+wasabi>=0.9.0,<0.10.0
+pydantic~=1.8.2
+pre-commit~=2.17.0
+
+
+
+# format
+black>=22.3.0,<22.4.0
+isort~=5.10.1
```

Comment:
Ruff [does not replace black](https://docs.astral.sh/ruff/faq/#is-ruff-compatible-with-black). I would install it here as well as you want your ide formatter to use the same version (and it will default to use the black in the environment).

Reply:
Agreed with Kenneth! Also, running them natively ensures:
* Alignment between IDE-formatter, CLI and CI
* Faster performance, since it doesn't need to activate a .venv (pre-commit installs each utility in its own .venv)